docs/src/tutorials/tut_setup.md

```diff
@@ -1,13 +1,320 @@
+
+# output
+CallbackSet{Tuple{}, ...}
+```
+
+Finally, we can start the simulation by solving the `ODEProblem`.
+We use the method `RDPK3SpFSAL35` of OrdinaryDiffEq.jl, which is a Runge-Kutta
+method with automatic (error based) time step size control.
+This method is usually a good choice for prototyping, since we do not have to
+worry about choosing a stable step size and can just run the simulation.
+For better performance, it might be beneficial to tweak the tolerances
+of this method or choose a different method that is more efficient for the
+respective simulation.
+You can find both approaches in our [example files](@ref examples).
+Here, we just use the method with the default parameters, and only disable
+`save_everystep` to avoid expensive saving of the solution in every time step.
+```jldoctest tut_setup; output = false, filter = r".*"s
+sol = solve(ode, RDPK3SpFSAL35(), save_everystep=false, callback=callbacks);
+
+# output
+
+```
+
+See [Visualization](@ref) for how to visualize the solution.
+
+## Replacing components with custom implementations
+
+If we would like to use an implementation of a component that is not available
+in TrixiParticles.jl, we can implement it ourselves within the simulation file,
+without ever cloning the TrixiParticles.jl repository.
+A good starting point is to check out the available implementations in
+TrixiParticles.jl, then copy the relevant functions to the simulation file
+and modify them as needed.
+
+### Custom smoothing kernel
+
+To implement a custom smoothing kernel, we define a struct extending
+`TrixiParticles.SmoothingKernel`.
+This abstract struct has a type parameter for the number of dimensions,
+which we set to 2 in this case.
+
+```jldoctest tut_setup; output = false
+struct MyGaussianKernel <: TrixiParticles.SmoothingKernel{2} end
+
+# output
+
+```
+This kernel is going to be an implementation of the Gaussian kernel with
+a cutoff for compact support.
+Note that the same kernel in a more optimized version is already implemented
+in TrixiParticles.jl as [`GaussianKernel`](@ref).
+
+In order to use our new kernel, we have to define three functions.
+`TrixiParticles.kernel`, which is the kernel function itself,
+`TrixiParticles.kernel_deriv`, which is the derivative of the kernel function,
+and `TrixiParticles.compact_support`, which defines the compact support of the
+kernel in relation to the smoothing length.
+The latter is relevant for determining the search radius of the neighborhood search.
+
+```jldoctest tut_setup; output = false
+function TrixiParticles.kernel(kernel::MyGaussianKernel, r, h)
+    q = r / h
+
+    if q < 2
+        return 1 / (pi * h^2) * exp(-q^2)
+    end
+
+    return 0.0
+end
+
+function TrixiParticles.kernel_deriv(kernel::MyGaussianKernel, r, h)
+    q = r * h
+
+    if q < 2
+        return 1 / (pi * h^2)  * (-2 * q) * exp(-q^2) / h
+    end
+
+    return 0.0
+end
+
+TrixiParticles.compact_support(::MyGaussianKernel, h) = 3 * h
+
+# output
+
```

Comment:
maybe add something like `plot(my_kernel)` to visualize the Kernel in 1D (or more fancy in 2D)?